pkg/metrics/azure_metrics.go

```diff
@@ -81,22 +82,35 @@ func (mc *MetricContext) ThrottledCount() {
 
 // Observe observes the request latency and failed requests.
 func (mc *MetricContext) Observe(rerr *retry.Error) {
-	apiMetrics.latency.WithLabelValues(mc.attributes...).Observe(
-		time.Since(mc.start).Seconds())
+	latency := time.Since(mc.start).Seconds()
+	apiMetrics.latency.WithLabelValues(mc.attributes...).Observe(latency)
 	if rerr != nil {
 		errorCode := rerr.ServiceErrorCode()
 		attributes := append(mc.attributes, errorCode)
 		apiMetrics.errors.WithLabelValues(attributes...).Inc()
 	}
+	mc.logLatency(latency, "error_code", rerr.ServiceErrorCode())
 }
 
 // ObserveOperationWithResult observes the request latency and failed requests of an operation.
 func (mc *MetricContext) ObserveOperationWithResult(isOperationSucceeded bool) {
```

Comment:
add a new parameter(`operation string`) in `ObserveOperationWithResult` and then in L104, write like `mc.logLatency(latency, operation, resultCode)`

Reply:
Those metadata would be logged (including the request -> request is the operation):

```go
metricLabels = []string{
		"request",         // API function that is being invoked
		"resource_group",  // Resource group of the resource being monitored
		"subscription_id", // Subscription ID of the resource being monitored
		"source",          // Operation source(optional)
	}

```